tests/experimental/wgit/test_api.py

```diff
@@ -0,0 +1,109 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
```

Comment:
Thanks for adding those top level (integration) tests!

I think we should also include unit test for each class, like test_repo.py, test_sha1_store.py, test_pygit.py etc. Basically, with unit test, we can test each APIs in separation and make bugs and regression much easier to catch.